dotnet/src/Microsoft.AutoGen/Agents/AgentBase.cs

```diff
@@ -31,6 +34,8 @@ protected AgentBase(IAgentContext context, EventTypes eventTypes)
 
     internal Task Completion { get; }
 
+    Abstractions.AgentId IAgentBase.AgentId => throw new NotImplementedException();
```

Comment:
why this?

Reply:
not sure why have two different scopes for AgentId.

Reply:
Looks like the entire purpose of the second `AgentId` (the one not from .proto) is to provide a constructor that not generated by grpc tools plus a pretty-print `toString` override.